app/Models/User.php

```diff
@@ -756,6 +791,9 @@ public function getStatus(): string
      */
     public function setStatusFor(Semester $semester, $status, $comment = null): User
     {
+        if ($semester==Semester::current() && $this->isTenant() && $status == SemesterStatus::ACTIVE) {
+            $this->setTenantUntilToSemesterEnd();
```

Comment:
I'd add a setTenantDateForTenantCollegists() User model function that includes the checks also. Just call that everywhere after changing any status/role.